// etch-stateless/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

exports["default"] = stateless;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function stateless(etch, renderFn) {
  if (typeof etch === 'function') {
    throw new Error("The first parameter to stateless should be the Etch library");
  }

  var EtchStateless = (function () {
    function EtchStateless(props, children) {
      _classCallCheck(this, EtchStateless);

      this.props = props;
      this.children = children;
      etch.initialize(this);
    }

    _createClass(EtchStateless, [{
      key: "update",
      value: function update(props, children) {
        this.props = props;
        this.children = children;
        return etch.update(this);
      }
    }, {
      key: "render",
      value: function render() {
        return renderFn(this.props, this.children);
      }
    }]);

    return EtchStateless;
  })();

  return EtchStateless;
}

module.exports = exports["default"];
